lib/util/itinerary.js

```diff
@@ -247,13 +247,21 @@ export function getLegBounds (leg) {
 
 export function routeComparator (a, b) {
   let aComp, bComp
-  if (a.sortOrder !== null && b.sortOrder !== null) {
+  // If route sort order field is available, default sort to these predefined
+  // values. NOTE: OTP/OBA defaults sort order values to -999, so instead of
+  // checking that the sort order is not null, we just check that the values are
+  // not equal. Technically, there could be cases where route_sort_order values
+  // are equivalent, but ideally they should be unique.
+  if (a.sortOrder !== b.sortOrder) {
     aComp = a.sortOrder
     bComp = b.sortOrder
   } else if (!isNaN(parseInt(a.shortName)) && !isNaN(parseInt(b.shortName))) {
+    // Otherwise, if both short names can be parsed as integers, use these
+    // numbers for sorting.
```

Comment:
There is a chance that two routes could have the exact same number value for shortName. In that case, I think it would make sense to defer to sorting based off of shortName/longName.

Reply:
The viewer doesn't distinguish by agency, does it?

Reply:
@binh-dam-ibigroup, it used to, but it's been commented out:

https://github.com/opentripplanner/otp-react-redux/blob/121066124e7804575e8dc3a73ad7209f5a755453/lib/components/viewers/route-viewer.js#L135-L140

Reply:
@binh-dam-ibigroup, it does handle agencies:
https://github.com/opentripplanner/otp-react-redux/blob/12d90349be3c7f68e041f422e1092ea79a32d77d/lib/components/viewers/route-viewer.js#L61-L65